sdk/storage/azure-storage-blob/azure/storage/blob/_models.py

```diff
@@ -1296,6 +1142,160 @@ def __init__(self, **kwargs):
         self.rules = kwargs.pop('rules', None)
 
 
+class BlobProperties(DictMixin):
+    """Blob Properties."""
+
+    name: str
+    """The name of the blob."""
+    container: str
+    """The container in which the blob resides."""
+    snapshot: str
+    """Datetime value that uniquely identifies the blob snapshot."""
+    blob_type: "BlobType"
+    """String indicating this blob's type."""
+    metadata: Dict[str, str]
+    """Name-value pairs associated with the blob as metadata."""
+    last_modified: "datetime"
+    """A datetime object representing the last time the blob was modified."""
+    etag: str
+    """The ETag contains a value that you can use to perform operations
+        conditionally."""
+    size: int
+    """The size of the content returned. If the entire blob was requested,
+        the length of blob in bytes. If a subset of the blob was requested, the
+        length of the returned subset."""
+    content_range: str
+    """Indicates the range of bytes returned in the event that the client
+        requested a subset of the blob."""
+    append_blob_committed_block_count: int
+    """(For Append Blobs) Number of committed blocks in the blob."""
+    is_append_blob_sealed: bool
+    """Indicate if the append blob is sealed or not.
+
+        .. versionadded:: 12.4.0"""
```

Comment:
Do we still want these? I left them in but can remove them if we think they are unnecessary / want to format them differently.